### ISSUE TITLE
Escape strings if pasted in to literal quotes or add command to escape HCL string literals

### ISSUE THREAD

**Issue by paultyng** [hashicorp/terraform-ls#431]
I'm not sure if this is a VS Code or LS thing, but it would be nice to be able to paste string values in to literal strings in HCL and have it be escaped somehow, or maybe as an operation on a string, I'm unsure what the idioms are that exist in other languages.

### Use-cases

Pasting raw JSON in to an HCL string literal.

### Attempted Solutions

Manual find and replace.

**Comment by paultyng** [comment]
Also, hey @radeksimko 👋🏻 😄

**Comment by radeksimko** [comment]
Hey @paultyng
Raw strings shouldn't need escaping when wrapped in heredoc notation in HCL, e.g.

```hcl
policy = <<POLICY
{"valid": "json"}
POLICY
```
which is a good practice anyway (compared to putting JSON in-line) to keep it readable.

However I'm assuming that you have raw JSON and you want to turn it into HCL's "object" representation, i.e. basically turn this:

```json
{
	"Version": "2012-10-17",
	"Statement": [{
		"Action": [
			"ec2:Describe*"
		],
		"Effect": "Allow",
		"Resource": "*"
	}]
}
```
into this
```hcl
jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Action = [
          "ec2:Describe*",
        ]
        Effect   = "Allow"
        Resource = "*"
      },
    ]
})
```

Am I right?

**Comment by paultyng** [comment]
In this case it wasn't JSON, I was just pasting in a complex password to a variable (I know it should probably not be in the *.tf 😂 ), so I had something like:

Clipboard: `abcd"efg`
HCL:
```terraform
variable "foo" {
  default = "<pasting in here>"
}
```

**Comment by paultyng** [comment]
You are correct though that if using multi-line JSON, a HEREDOC is much much easier.

**Comment by radeksimko** [comment]
> In this case it wasn't JSON, I was just pasting in a complex password to a variable (I know it should probably not be in the *.tf 😂 ), so I had something like:

🙈  :joysob:

Jokes aside I'm not fully convinced that helping people escape _in-line_ JSON is a good idea and I'm not sure how reliably would escaping for arbitrary strings work and how we could implement it since pasting unescaped string into HCL config has a potential of making parts of it invalid/unparseable.

However providing conversion from JSON to HCL object is a good idea - I reckon that would also help syntax highlighting. I think we could assign a code lens action to any valid JSON string, so we can keep tracking this under LS.

**Comment by paultyng** [comment]
👍🏻 works for me, any way to more easily handle this kind of stuff sounds great.
